src/main/java/com/acme/dbo/txlog/Facade.java

```diff
@@ -1,11 +1,41 @@
 package com.acme.dbo.txlog;
 
 public class Facade {
+    public static boolean isDecorated;
+
+    private static void decorateLogPrimitive(String message) {
+        if (Facade.isDecorated) {
+            System.out.println("primitive: " + message);
+        } else {
+            System.out.println(message);
+        }
+    }
+
     public static void log(int message) {
-        System.out.println("primitive: " + message);
+        Facade.decorateLogPrimitive(String.valueOf(message));
+    }
+
+    public static void log(boolean message) {
+        Facade.decorateLogPrimitive(String.valueOf(message));
+    }
+
+    public static void log(char message) {
```

Comment:
Full code duplication with decorateLogPrimitive()

Reply:
What can be done here if these are two different methods for boolean and Integer?

Reply:
Extract methods (internal components) from methods for boolean and integer. You can deconstruct methods to see what parts it consist of. And then you can reuse these parts.
If this is too complex for now - don't worry, we'll discuss it at webinar session.

Reply:
Hope it's resolved but not ideal.